contracts/eosiolib/multi_index.hpp

```diff
@@ -562,15 +562,14 @@ class multi_index
 
          //using malloc/free here potentially is not exception-safe, although WASM doesn't support exceptions
          void* buffer = max_stack_buffer_size < size_t(size) ? malloc(size_t(size)) : alloca(size_t(size));
+         struct { 
+            struct S { ~S()  { if (buf) free(buf); } void *buf; } s; 
+         } __freemem{max_stack_buffer_size<size_t(size)?buffer:0};          
 
          db_get_i64( itr, buffer, uint32_t(size) );
 
          datastream<const char*> ds( (char*)buffer, uint32_t(size) );
 
-         if ( max_stack_buffer_size < size_t(size) ) {
-            free(buffer);
-         }
-
          auto itm = std::make_unique<item>( this, [&]( auto& i ) {
```

Comment:
Changes should be submitted for this to the https://github.com/EOSIO/eosio.wasmsdk repo.

I would recommend just moving the free down and not use the RAII approach.